doc/kubernetes/modules/ROOT/pages/running/deployments/active-passive-sync.adoc

```diff
@@ -0,0 +1,214 @@
+| No data loss
+| Less than one minute
+
+| Infinispan node
+| Multiple Infinispan instances run in each datacenter. If one instance fails, it takes a few seconds to the other nodes to notice the change. Sessions are stored in at least two Infinispan nodes, so a single node failure doesn't lead to data loss.
+| No data loss
+| Less than one minute
+
+| Connectivity Infinispan
+| If the connectivity between the two datacenters is lost, session information can't be sent to the other datacenter.
+Incoming requests might receive an error message or are delayed for some seconds.
+The primary site marks the secondary site offline, and will stop sending data to the secondary.
+The setup is degraded until the connection is restored and the session data is re-synchronized to the secondary datacenter.
+| No data loss ^3^
+| Less than one minute
+
+| Connectivity Database
+| If the connectivity between the two datacenters is lost, the synchronous replication will fail, and it might take some time for the primary site to mark the secondary offline.
+Some requests might receive an error message or are delayed for some seconds.
+Manual operations might be necessary depending on the database.
+| No data loss ^3^
+| Seconds to minutes (depending on the database)
+
+| Primary Datacenter
+| If none of the Keycloak nodes is available, the loadbalancer will detect the outage and redirect the traffic to the secondary site.
+Some requests might receive an error message while the loadbalancer hasn't detected the primary datacenter failure.
+The setup will be degraded until the primary site is back up and the session state has been manually synced from the secondary to the primary site.
+| No data loss^3^
+| Less than one minute
+
+| Secondary Datacenter
+| If the secondary datacenter is not available, it will take a moment for the primary Infinispan and database to mark the secondary datacenter offline.
+Some requests might receive an error message while the detection takes place.
+Once the secondary datacenter is up again, the session state needs to be manually synced from the primary site to the secondary site.
+| No data loss^3^
+| Less than one minute
+
+|===
+
+^1^: Recovery point objective, assuming all parts of the setup were healthy at the time this occurred. +
+^2^: Recovery time objective. +
+^3^: Manual operations needed to restore the degraded setup.
+
+The statement "`No data loss`" depends on the setup not being degraded from previous failures, which includes completing any pending manual operations to resynchronize the state between the datacenters.
+
+=== Known limitations
+
+==== Upgrades
+
+* On Keycloak major version upgrade, all session data (except offline session) might be lost as the data format is not guaranteed to be compatible between major versions.
+* On any Infinispan upgrade (major, minor or patch), all session data might be lost as cross-DC communication is guaranteed to work only between identical versions of Infinispan.
```

Comment:
Do we want the data to survive Infinispan upgrades? Customers may configure persistence (file-store for example) to survive cluster upgrade/restart.

Reply:
Yes we want. Still it might be a future milestone. Persistence might solve some of it, still I would assume it would mean a downtime as both DCs would need to use the same patch-level version of Infinispan to communicate?

Reply:
Yes. RollingUpgrades should allow to upgrade without downtime but that isn't properly tested with cross-dc.